### PR TITLE
cdc: limit scan speed

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

There will be a series of PRs aims to fix the TiKV CDC OOM issue (too many pending send messages are buffered in memory).

To resolve the issue, we have two strategies,
1. Limit the speed of producing messages during the initial scan stage. #9948 
2. Set an upper bound of pending send messages. #9885 

The series of PRs adopts both strategies.

This PR cherry-picks #9948 to the master branch. Unfortunately, the master branch diverges release-5.0 and it is hard to pick, so I plan to revert flow control #9885 temporarily and will bring it back step by step in other PRs.

Plan:
- take1: cherry-pick speed limiter <-- we are here.
- take2: set an upper bound of internal channel used in CDC.
- take3: add memory quota to CDC and kill connection if it uses too much memory.
- take4: cherry-pick other bugfixes in #9885 

See full changes: https://github.com/tikv/tikv/compare/master...overvenus:cdc/flowcontrol/all

Cc #8168 #9996

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note.

### CODE REVIEW

**Review by 5kbpers — APPROVED** [review]
LGTM

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -543,96 +526,61 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
...
             build_resolver: is_new_delegate,
-            real_time_start_ts: None,
-            advanced_flow_control_enabled,
         };
 
         let (cb, fut) = tikv_util::future::paired_future_callback();
         let scheduler = self.scheduler.clone();
         let deregister_downstream = move |err| {
-            let deregister = if is_new_delegate {
-                Deregister::Region {
```
Comment (liuzix) [inline components/cdc/src/endpoint.rs:574]: Maybe we need this

**Pull request merged**